src/hooks/video/agora/useAgoraRemotes.ts

```diff
@@ -29,7 +29,16 @@ export const useAgoraRemotes: ReactHook<
   );
 
   useEffect(() => {
-    if (!client) return;
+    if (!client) {
+      setClient(
+        AgoraRTC.createClient({
+          codec: "h264",
+          mode: "rtc",
+        })
+      );
+
+      return;
+    }
```

Comment:
Can this cause a loop? What if there is an error in setting the client?

Reply:
@mike-lvov there shouldn't be the loop as we get the object with client in any case. As a variant, I can put here the creating of client in useEffect with empty dependencies. After the calling to createClient function need to check if we have client or not, if not so use Bugsnag. I didn't find any other extra API, only this one but if I understand correctly we are talking about the connection of the client to the channel and all info is sending to the servers of Agora. https://docs.agora.io/en/All/API%20Reference/web/modules/agorartc.logger.html#enablelogupload